### ISSUE TITLE
ZFS hole

### ISSUE THREAD

**Issue by zheliazkov** [openzfs/zfs#12308]
### Discussed in https://github.com/openzfs/zfs/discussions/12307

<div type='discussions-op-text'>

<sup>Originally posted by **zheliazkov** July  1, 2021</sup>
Hello,

I'll be glad to get information/help about the following issue:

1. I've pool with with multiple raid-z2 vdevs and one log device
2. I wanted to replace the log device and removed it from the pool
3. The status of the zpool is currently ok
4. Now I want to put another devices as a read and write cache but when I try dry-run (-n) it shows some hole, and the log device is not in the log "section" of the output

```
# zpool add -n stor log /dev/disk/by-id/nvme-Corsair_MP400_....
would update 'stor' to the following configuration:
        stor
          raidz2-0
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
          raidz2-1
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
          hole-2
          raidz2-3
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_....
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
            scsi-SHGST_HUH728080AL4200_...
          nvme-Corsair_MP400_...
```
The questions are:
1. Why this hole appears in the dry-run and not in the current status and what it is? Is it a problem?
2. Why the new dev is not in a log "section"?

Thanks in advance.

BR,
Plamen</div>

**Comment by behlendorf** [comment]
As mentioned in this https://github.com/openzfs/zfs/discussions/12307#discussioncomment-951269 this issue was resolved in OpenZFS 2.0.2.

**Comment by zheliazkov** [comment]
Hi @behlendorf,
Thanks for the info.

Best Regards,
Plamen
